src/AzureStoragePersistenceSagaExporter/CloudTableExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿namespace Particular.AzureStoragePersistenceSagaExporter
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.WindowsAzure.Storage.Table;
+
+    static class CloudTableExtensions
+    {
+        public static async Task<IList<T>> ExecuteQueryAsync<T>(this CloudTable table, TableQuery<T> query, int take = int.MaxValue, CancellationToken ct = default(CancellationToken)) where T : ITableEntity, new()
```

Comment:
This would be a perfect use case for async enumerable because then we don't need to materialise all into mem and can pull items as they come